src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,14 +1,44 @@
-import React from 'react';
+import { useDispatch } from 'react-redux';
+import { useAppSelector } from '../../app/hooks';
+import { actions as filterActions } from '../../features/filter';
+
+import { StatusTodos } from '../../types/StatusTodos';
+import { ShowTodos } from '../../types/ShowTodos';
 
 export const TodoFilter: React.FC = () => {
+  const dispatch = useDispatch();
+  const query = useAppSelector(state => state.filter.query);
+
+  const handleFilterStatus = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    switch (event.target.value) {
+      case StatusTodos.Active:
+        dispatch(filterActions.setStatus(ShowTodos.Active));
+        break;
+
+      case StatusTodos.Completed:
+        dispatch(filterActions.setStatus(ShowTodos.Completed));
+        break;
+
+      default:
+        dispatch(filterActions.setStatus(ShowTodos.All));
+        break;
+    }
+  };
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    dispatch(filterActions.setQuery(event.target.value));
+  };
+
+  const clearQuery = () => dispatch(filterActions.setQuery(null));
+
```

Comment:
You are setting the query to null, which is not a good practice. Instead, you should set the initial value to an empty string.

Reply:
Refactor